src/engraving/dom/shape.cpp

```diff
@@ -297,6 +297,30 @@ double Shape::bottom() const
     return dist;
 }
 
+double Shape::rightMostEdgeAtHeight(double yAbove, double yBelow) const
+{
+    double edge = -100000.0;
+    for (const ShapeElement& sh : *this) {
+        if (sh.bottom() > yAbove && sh.top() < yBelow) {
+            edge = std::max(edge, sh.right());
+        }
+    }
+
+    return edge;
+}
+
+double Shape::leftMostEdgeAtHeight(double yAbove, double yBelow) const
+{
+    double edge = 100000.0;
+    for (const ShapeElement& sh : *this) {
+        if (sh.bottom() > yAbove && sh.top() < yBelow) {
```

Comment:
Is there any possibility we'll never pass this condition? If so how 100000.0 value will affect the layout/drawing?

Reply:
see #20656, @alexpavlov96 you may want to pick that commit too